keel-core/src/main/kotlin/com/netflix/spinnaker/keel/telemetry/TelemetryEvent.kt

```diff
@@ -50,3 +50,7 @@ data class ArtifactCheckTimedOut(
   val name: String,
   val deliveryConfigName: String?
 ) : TelemetryEvent()
+
+data class ArtifactVersionVetoed(
+  val artifactReference: String,
```

Comment:
IMHO, we should stop using `artifactReference` alone as an identifier, and couple it with `artifactType`. Especially now that we can specify artifacts based on branches, I can easily see a case where you might have a frontend NPM package coming from master that you label `master`, and a backend Docker/Debian that you also label `master`. The difference is the type.

Reply:
@luispollo This is just for a telemetry event - because I want to add a tag here. I think you're right that the reference isn't useful (and it's not enough to identify as you pointed out). If an artifact is vetoed, what information do you want? Maybe the application is best here? We will have to dig into logs to see anything useful regardless.

Reply:
Well, even in Atlas, the problem I mentioned would come up, so I think it'd be worth adding the type here. Other than that, my comment was just because I see us propagating this pattern of using _just_ the reference to identify artifacts (even in repository methods), and it'll be harder to fix the more we add.

Reply:
I switched to application because I think that at a glance we want to know which app is having artifacts vetoed. Then we can dive into the logs to find out more info

Reply:
You beat me to it! I was just about to suggest that. Good call!

Also, I take back what I said about artifact references. The whole point of those was to be able to use a single string value in a resource spec to point to an artifact, so it makes sense for them to be globally unique in the delivery config! So, in order to reason about an artifact in logs/metrics, we need the application and artifact reference as you did.